### ISSUE TITLE
Nif execution error on agent 

### ISSUE THREAD

**Issue by ChiralMedia** [basho-labs/riak-mesos-tools#18]
Mesos 0.28.1  DC/OS 1.7  Riak framework 0.1.1

After reinstallation of framework riak-mesos executor crashes.

6:46:11.493 [warning] The on_load function for module nif_executor returned {error,{load_failed,"Failed to load NIF library /var/lib/mesos/slave/slaves/e276daa8-56e7-473f-b362-440030666973-S1/frameworks/81850ec1-edb2-4feb-ab76-a689a3e11312-0000/executors/riak.f834e50c-45d7-11e6-9aea-0242cd3482b0/runs/b73123d6-4204-46fd-94b3-6fbe0170916d/riak_mesos_executor/bin/../lib/erlang_mesos-0.23.0/priv/executor: 'libmesos-0.28.1.so: cannot open shared object file: No such file or directory'"}}
(riak_mesos_executor@127.0.0.1)1> 16:46:11.493 [error] CRASH REPORT Process executor with 0 neighbours exited with reason: call to undefined function nif_executor:init(<0.138.0>) in gen_server:init_it/6 line 328

http://localhost:9000 shows Riak explorer, but it is unable to see the local node or any cluster however DC/OS reports all instances as healthy.

Output attached.

[AgentCrashIssue.txt](https://github.com/basho-labs/riak-mesos-tools/files/363760/AgentCrashIssue.txt)

**Comment by sanmiguel** [comment]
Hi there!

I see you've raised several issues here, but I think they may all have a common cause, so let's address that first and take it from there.

I think you have installed Riak from the package found in the DCOS Universe. Unfortunately this version is in the process of being replaced with a completely rebuilt framework. In order to use the most recent version, I recommend first uninstalling the Riak package you have now, and then:

```
$ dcos package repo add --index=0 RMF https://github.com/basho-labs/riak-mesos-dcos-repo/archive/dcoscli-v0.4.x.zip
$ dcos package search riak
NAME  VERSION  SELECTED  FRAMEWORK  DESCRIPTION
riak  1.0.0    False     True       A distributed NoSQL key-value data store that offers high availability, fault tolerance, operational simplicity, and scalability.
```

Assuming you now see riak v1.0.0 available, you should be able to install it thus (and assuming your riak-mesos config file is at `/etc/riak-mesos/config.json`) :

```
$ dcos package install --options=/etc/riak-mesos/config.json riak
```

**Comment by ChiralMedia** [comment]
Hello,
 
Thanks for the quick response.
We've updated the repo, the results of the install are below. I've attached the config,json we're using in case it's of any use.
 
Regards,
David
 
[u1@bootstrapper ~]$ dcos package search riak
NAME VERSION SELECTED FRAMEWORK DESCRIPTION 
riak 1.0.0 False True A distributed NoSQL key-value data store that offers high availability, fault... 
[u1@bootstrapper ~]$ 
[u1@bootstrapper ~]$ dcos package install --options=/etc/riak-mesos/config.json riak
The Riak framework should have at least 1GB of RAM and 0.5 CPUs to perform successfully.
Continue installing? [yes/no] yes
Package file [marathon.json] is not JSON: expected json value got h (line 24, column 1620)
[u1@bootstrapper ~]$
  
  --------- Original Message ---------
 
 Subject: Re: [basho-labs/riak-mesos-tools] Nif execution error on agent (#18)
